chain/network/src/peer/codec.rs

```diff
@@ -78,19 +78,20 @@ impl Decoder for Codec {
     type Error = Error;
 
     fn decode(&mut self, buf: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
-        if buf.len() < 4 {
+        let len_buf = match buf.get(..4).and_then(|s| <[u8; 4]>::try_from(s).ok()) {
```

Comment:
Alternative way to write this could be:

```rust
let len = match buf {
    [a, b, c, d, ..] => u32::from_le_bytes([a, b, c, d]) as usize,
    _ => return Ok(None),
};
```